backend/src/utils/features.ts

```diff
@@ -1,10 +1,16 @@
 import * as fs from 'fs';
 import * as path from 'path';
+import { getDashboardConfig } from './resourceUtils';
 
 export const getComponentFeatureFlags = (): { [key: string]: string } => {
   const normalizedPath = path.join(__dirname, '../../../data/features.json');
   try {
-    return JSON.parse(fs.readFileSync(normalizedPath, 'utf8'));
+    const features = JSON.parse(fs.readFileSync(normalizedPath, 'utf8'));
```

Comment:
no synchronous anything, ever.

Reply:
@maroroman don't worry about this, I'll address this separately

Reply:
I don't think I'd mess with features at all in this file since it's static json.  I would just get the configmap in routes/api/components and filter there based on the configmap.